screeninfo/enumerators/windows.py

```diff
@@ -64,28 +64,38 @@ def callback(monitor: T.Any, dc: T.Any, rect: T.Any, data: T.Any) -> int:
     # multiple monitors have different DPIs.
     ctypes.windll.shcore.SetProcessDpiAwareness(2)
 
+    # Type of HWND resolves to void *
+    # HWND -> HANDLE -> PVOID -> void *
+    # See https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types
+    hwnd = ctypes.c_void_p(None)
     # On Python 3.8.X GetDC randomly fails returning an invalid DC.
     # To workaround this request a number of DCs until a valid DC is returned.
     for retry in range(100):
         # Create a Device Context for the full virtual desktop.
-        dc_full = ctypes.windll.user32.GetDC(None)
-        if dc_full > 0:
+        # On failure, GetDC returns NULL (type: HDC)
+        # HDC eventually resolves to void *
+        # HDC -> HANDLE -> PVOID -> void *
+        # See https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types
+        GetDC = ctypes.windll.user32.GetDC
+        GetDC.restype = ctypes.c_void_p
+        dc_full = GetDC(hwnd)
```

Comment:
Is it necessary to convert `None` to `c_void_p` by hand?

Reply:
Not necessary. I did this for two reasons:
- Using the same variable to when getting the DC and when releasing makes helps to clarify that it supposed to be the same value. 
- Specifying the types make it clear what is expected by the function, and that the type has been checked to be correct and intentionally set. Otherwise it's just relying on ctypes to guess and get it right.